pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -0,0 +1,36 @@
+package clusterapi
+
+import (
+	"context"
+
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	gcptypes "github.com/openshift/installer/pkg/types/gcp"
+)
+
+type Provider struct {
+}
+
+var _ clusterapi.PreProvider = (*Provider)(nil)
+var _ clusterapi.IgnitionProvider = (*Provider)(nil)
+var _ clusterapi.InfraReadyProvider = (*Provider)(nil)
+var _ clusterapi.PostProvider = (*Provider)(nil)
+
+func (p Provider) Name() string {
+	return gcptypes.Name
+}
+
+func (p Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error {
```

Comment:
nit: as we're not saving any state in the Provider, you might consider switching to use the unnamed provider:

```suggestion
func (_ Provider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error {
```
I'm fine with saving state in the provider,  but we're not doing it yet.